images/router/haproxy/conf/haproxy-config.template

```diff
@@ -620,7 +620,8 @@ backend {{ genBackendNamePrefix $cfg.TLSTermination }}:{{ $cfgIdx }}
               {{- range $idx, $endpoint := processEndpointsForAlias $cfg $serviceUnit (env "ROUTER_BACKEND_PROCESS_ENDPOINTS" "") }}
   server {{ $endpoint.ID }} {{ $endpoint.IP }}:{{ $endpoint.Port }} cookie {{ $endpoint.IdHash }} weight {{ $weight }}
                 {{- if (eq $cfg.TLSTermination "reencrypt") }} ssl
-                  {{- if not (isTrue $router_disable_http2) }} alpn h2,http/1.1
+                  {{- if eq $endpoint.AppProtocol "h2c" }} proto h2
```

Comment:
How does this work (or does this work) if HTTP/2 is disabled?

Reply:
Adding `proto h2` to the `server` stanza affects the protocol used for the connection from HAProxy to the backend and doesn't affect the protocol used for the connection from the client to the frontend, which is governed by the `sslbindconf` in the certificate map.